### PR TITLE
Create moderation request on publish

### PR DESCRIPTION
**Pull request by alejandromumo**
closes https://github.com/inveniosoftware/invenio-requests/issues/344

Needs:
- https://github.com/inveniosoftware/invenio-requests/pull/345

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -297,6 +297,10 @@ def app_config(app_config, mock_datacite_client):
     app_config["USERS_RESOURCES_SERVICE_SCHEMA"] = NotificationsUserSchema
 
     app_config["RDM_RESOURCE_ACCESS_TOKENS_ENABLED"] = True
+
+    # Users are verified by default. This will disable the automatic creation of moderation requests after publishing a record.
+    # When testing unverified users, there is a "unverified_user" fixture for that purpose.
+    app_config["ACCOUNTS_DEFAULT_VERIFIED_STATUS"] = True
```
Comment (alejandromumo) [inline tests/conftest.py:303]: This simplifies testing. 

That way, after `publish` there are no moderation requests created in tests and it's easier to control which user is verified / not-verified (I created explicit users for that purpose).

File: invenio_rdm_records/services/components/verified.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class ContentModerationComponent(ServiceComponent):
+    """Service component for content moderation."""
+
+    def publish(self, identity, draft=None, record=None):
+        """Create a moderation request if the user is not verified."""
+        is_verified = record.parent.is_verified
+
+        # Owner might not resolve to a User (e.g. when the record was created by system)
+        owner_exists = record.parent.access.owner.owner_id
```
Comment (alejandromumo) [inline invenio_rdm_records/services/components/verified.py:24]: Try to identify if it's the system.

File: tests/requests/test_user_moderation_actions.py [hunk 3]
```diff
@@ -5,37 +5,59 @@
...
 
     # Create a new version of the record
-    new_version = records_service.new_version(uploader.identity, id_=record.id)
-    records_service.update_draft(uploader.identity, new_version.id, minimal_record)
-    new_record = records_service.publish(identity=uploader.identity, id_=new_version.id)
+    new_version = records_service.new_version(unverified_user.identity, id_=record.id)
+    records_service.update_draft(
+        unverified_user.identity, new_version.id, minimal_record
+    )
+    # Since tasks are executed synchronously in tests, the db session that is rolledback inside the session
```
Comment (zzacharo) [inline tests/requests/test_user_moderation_actions.py:52]: rephrase a bit to depict what is wrong?

File: tests/requests/test_user_moderation_actions.py [hunk 4]
```diff
@@ -47,11 +69,17 @@ def test_user_moderation_approve(
 
     assert is_verified == False
 
-    moderation_request = mod_request_create(uploader.id)
+    # Approve the user
```
Comment (zzacharo) [inline tests/requests/test_user_moderation_actions.py:72]: outdated?

**Review by zzacharo — APPROVED** [review]

**Pull request merged**